### PR TITLE
US002 Intup Modal Controller

### PR DESCRIPTION
**Pull request by Zeroya**
Add input, modal window and logic for adding items , some styles added

### CODE REVIEW

File: src/components/ModalWindow/ModalWindow.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  const dispatch = useAppDispatch();
+
+  const handleSubmit = (e: React.FormEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+    if (/^[A-Za-zА-Яа-яЁё0-9\s]+$/.test(input.message) && /[0-9]+/.test(input.data) && /[0-9]+/.test(input.expData)) {
+      dispatch(addModalTodo(input));
+      setInput({});
+      setShow(false);
+    }
+  }
```
Comment (Kost927) [inline src/components/ModalWindow/ModalWindow.tsx]: check eslint or prettier configuration. Coz you don't have semicolons after methods.

File: src/components/ModalWindow/ModalWindow.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import Button from 'react-bootstrap/Button';
+import Form from 'react-bootstrap/Form';
+import Modal from 'react-bootstrap/Modal';
+import { useAppDispatch } from '../../hooks/hooks';
+import { addModalTodo } from '../../store/reducers/TodosSlice';
+import s from "./ModalWindow.module.scss";
+
+const ModalWindow: React.FC = () => {
+
+  const [input, setInput] = useState<any>({});
```
Comment (Kost927) [inline src/components/ModalWindow/ModalWindow.tsx]: type "any"

File: src/components/TodoItem/TodoItem.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import { ITodo } from '../../models/ITodo';
+import s from "./TodoItem.module.scss";
+
+const TodoItem: React.FC<{ todo: ITodo }> = (props) => {
+
+  return (
+    <li className={s.todoItem}>
+      <input type="checkbox" />
+      <div className={s.todoItem_message}>
+        <p>{props.todo.message}</p>
```
Comment (Kost927) [inline src/components/TodoItem/TodoItem.tsx]: use destructuring to not write "props" all the time

File: src/components/ModalWindow/ModalWindow.tsx (hunk 2 above)
Comment (Zeroya) [reply src/components/ModalWindow/ModalWindow.tsx]: Yes, I added interface first but for some reason it didn't want to work through this I had to add 'any' after many tries

File: src/components/TodoItem/TodoItem.module.scss [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  p {
+    font-size: 14px;
+    margin: 0;
+    padding-top: 1em;
+    padding-bottom: 1em;
+  }
+
+  &_message {
+    width: 5em;
+    //overflow-x:hidden;
```
Comment (Kost927) [inline src/components/TodoItem/TodoItem.module.scss]: remove commented code

File: src/store/reducers/TodosSlice.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  name: 'todos',
+
+  initialState,
+  reducers: {
+    addTodo: (state, action: PayloadAction<string>) => {
+      state.todos.push({
+        id: uuidv4(),
+        message: action.payload,
+        completed: false,
+        data: new Date().toLocaleDateString().split(".").join(".") + ' ' + new Date().toLocaleTimeString().slice(0, -3),
```
Comment (Kost927) [inline src/store/reducers/TodosSlice.ts]: you can create a "utils" folder to put there all these calculations with dates.
Field name "data" is it a "date"? not clear.

File: src/store/reducers/TodosSlice.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        dataExpiration:new Date().toLocaleDateString().split(".").map((el,i) => i === 0 ? Number(el)+1: el).join(".") + ' ' + new Date().toLocaleTimeString().slice(0, -3),
+      })
+    },
+    addModalTodo: (state, action: PayloadAction<UserData>) => {
+      state.todos.push({
+        id: uuidv4(),
+        message: action.payload.message,
+        completed: false,
+        data: action.payload.data.split("").map(el => el === "T" ? " " : el === "-" ? "." : el).join("").slice(0,10).split(".").reverse().join('.').concat(" ").concat(action.payload.data.slice(11,16)),
+        dataExpiration:action.payload.expData.split("").map(el => el === "T" ? " " : el === "-" ? "." : el).join("").slice(0,10).split(".").reverse().join('.').concat(" ").concat(action.payload.expData.slice(11,16)),
```
Comment (Kost927) [inline src/store/reducers/TodosSlice.ts]: you can create a "utils" folder to put there all these calculations with dates. So you will have smth like this:
date: getCreationDate(action.payload), where getCreationDate() will return your date value
Field name "data" is it a "date"? not clear.

Comment (Zeroya) [reply src/store/reducers/TodosSlice.ts]: Understood, I will create a separate folder for calculations. about data , you are right, I confused with date

File: src/store/reducers/TodosSlice.ts (hunk 5 above)
Comment (Zeroya) [reply src/store/reducers/TodosSlice.ts]: yes, date)

File: src/components/ModalWindow/ModalWindow.tsx (hunk 2 above)
Comment (Kost927) [reply src/components/ModalWindow/ModalWindow.tsx]: make an initial state then, coz empty object is not your input state, or read how to use "as" {} as InputState

File: src/components/ModalWindow/ModalWindow.tsx (hunk 1 above)
Comment (Zeroya) [reply src/components/ModalWindow/ModalWindow.tsx]: I checked and fixed it, earlier for some reason all my plugins were removed from vs code, now I installed prettier, there should be no more such problems

File: src/components/ModalWindow/ModalWindow.tsx (hunk 2 above)
Comment (Zeroya) [reply src/components/ModalWindow/ModalWindow.tsx]: I made initial state, it works

File: src/components/TodoItem/TodoItem.tsx (hunk 3 above)
Comment (Zeroya) [reply src/components/TodoItem/TodoItem.tsx]: Understand, remade this moment for destructuring and then will do the same way.

File: src/components/TodoItem/TodoItem.module.scss (hunk 4 above)
Comment (Zeroya) [reply src/components/TodoItem/TodoItem.module.scss]: Fixed

File: src/store/reducers/TodosSlice.ts (hunk 5 above)
Comment (Zeroya) [reply src/store/reducers/TodosSlice.ts]: rename all data to date

File: src/store/reducers/TodosSlice.ts (hunk 6 above)
Comment (Zeroya) [reply src/store/reducers/TodosSlice.ts]: Create folder utils for calculations and rename data

**Review by Kost927 — APPROVED** [review]

**Pull request merged**